internal/lifecycle/readme_test.go

```diff
@@ -0,0 +1,438 @@
+				"project:region:instance",
+			},
+		},
+		cmds: []*exec.Cmd{
+			exec.Command("gcloud", "--quiet", "run", "services", "update", "unique_service_name", "--add-cloudsql-instances=project:region:instance"),
+		},
+		serviceName: "unique_service_name",
+		gcrURL:      "gcr.io/unique/tag",
+	},
+	{
+		in: "[//]: # ({sst-run-unix})\n" +
+			"```\n" +
+			"gcloud run services update hello_world --add-cloudsql-instances=\\\n" +
+			"${TEST_CLOUD_SQL_CONNECTION}\n" +
+			"```\n",
+		codeBlocks: []codeBlock{
+			[]string{
+				"gcloud run services update hello_world --add-cloudsql-instances=\\",
+				"${TEST_CLOUD_SQL_CONNECTION}",
+			},
+		},
+		cmds: []*exec.Cmd{
+			exec.Command("gcloud", "--quiet", "run", "services", "update", "unique_service_name", "--add-cloudsql-instances=project:region:instance"),
+		},
+		env: map[string]string{
+			"TEST_CLOUD_SQL_CONNECTION": "project:region:instance",
+		},
+		serviceName: "unique_service_name",
+		gcrURL:      "gcr.io/unique/tag",
+	},
+}
+
+func TestToCommands(t *testing.T) {
+	for i, tc := range tests {
+		err := setEnv(tc.env)
+		if err != nil {
+			t.Errorf("#%d: setEnv: %#v", i, err)
+
+			err = unsetEnv(tc.env)
+			if err != nil {
+				t.Errorf("#%d: unsetEnv: %#v", i, err)
+			}
+
+			continue
+		}
+
+		equalE := true
+		var cmds []*exec.Cmd
+		for j, codeBlock := range tc.codeBlocks {
+			h, err := codeBlock.toCommands(tc.serviceName, tc.gcrURL)
+			equalE = equalE && equalError(err, tc.toCommandsErr)
+			if !equalE {
+				t.Errorf("#%d.%d: error mismatch\nhave: %v\nwant: %v", i, j, err, tc.toCommandsErr)
+			}
+
+			cmds = append(cmds, h...)
+		}
+
+		if equalE && !reflect.DeepEqual(cmds, tc.cmds) {
+			t.Errorf("#%d: result mismatch\nhave: %#+v\nwant: %#+v", i, cmds, tc.cmds)
+		}
+
+		err = unsetEnv(tc.env)
+		if err != nil {
+			t.Errorf("#%d: unsetEnv: %#v", i, err)
+		}
+	}
+}
+
+func TestExtractLifecycle(t *testing.T) {
+	for i, tc := range tests {
+		err := setEnv(tc.env)
+		if err != nil {
+			t.Errorf("#%d: setEnv: %#v", i, err)
+
+			err = unsetEnv(tc.env)
+			if err != nil {
+				t.Errorf("#%d: unsetEnv: %#v", i, err)
+			}
+
+			continue
+		}
+
+		s := bufio.NewScanner(strings.NewReader(tc.in))
+		var c []*exec.Cmd
+		c, err = extractLifecycle(s, tc.serviceName, tc.gcrURL)
+
+		eE := equalError(err, tc.extractLifecycleErr)
+		if !eE {
+			t.Errorf("#%d: error mismatch\nhave: %v\nwant: %v", i, err, tc.extractLifecycleErr)
+		}
+
+		if eE && !reflect.DeepEqual(c, tc.cmds) {
+			t.Errorf("#%d: result mismatch\nhave: %#+v\nwant: %#+v", i, c, tc.cmds)
+		}
+
+		err = unsetEnv(tc.env)
+		if err != nil {
+			t.Errorf("#%d: unsetEnv: %#v", i, err)
+		}
+	}
+}
+
+func TestExtractCodeBlocks(t *testing.T) {
+	for i, tc := range tests {
+		s := bufio.NewScanner(strings.NewReader(tc.in))
+
+		h, err := extractCodeBlocks(s)
+		if !equalError(err, tc.extractCodeBlocksErr) {
+			t.Errorf("#%d: error mismatch\nhave: %v\nwant: %v", i, err, tc.extractCodeBlocksErr)
```

Comment:
Go has a common idiom for this: `want x, got y`

Reply:
Resolved offline, thank you.

Reply:
Resolved offline, thank you.